### PR TITLE
Extract snakeyaml to an API plugin

### PR DESCRIPTION
**Pull request by escoem**
Fixes https://github.com/jenkinsci/configuration-as-code-plugin/issues/1314

snakeyaml is now provided by a library plugin [Snakeyaml API Plugin](https://github.com/jenkinsci/snakeyaml-api-plugin) instead of the shaded version in the module https://github.com/jenkinsci/configuration-as-code-plugin/tree/master/snakeyaml

Code changes are related to the module removal (as the snakeyaml package is now the real one)

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [X] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [X] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -13,7 +13,6 @@
   <packaging>pom</packaging>
 
   <modules>
-    <module>snakeyaml</module>
```
Comment (timja) [inline pom.xml]: snakeyaml/{pom.xml,README.md} appear to still be present

**Comment by timja** [comment]
The only reference outside this plugin to snakeyaml I found was in the gradle-plugins tests so looks safe to remove,
https://github.com/search?q=org%3Ajenkinsci+io.jenkins.plugins.casc.snakeyaml&type=Code

File: test-harness/pom.xml [hunk 2]
```diff
@@ -56,6 +56,12 @@
       <groupId>com.fasterxml.jackson.dataformat</groupId>
       <artifactId>jackson-dataformat-yaml</artifactId>
       <version>${jackson.version}</version>
+      <exclusions>
+        <exclusion>
+          <groupId>org.yaml</groupId>
+          <artifactId>snakeyaml</artifactId>
```
Comment (amuniz) [inline test-harness/pom.xml]: :+1: I got bitten by this somewhere else!

File: pom.xml (hunk 1 above)
Comment (amuniz) [reply pom.xml]: Yeah, the whole module should be deleted.

**Comment by escoem** [comment]
> The only reference outside this plugin to snakeyaml I found was in the gradle-plugins tests so looks safe to remove,
> https://github.com/search?q=org%3Ajenkinsci+io.jenkins.plugins.casc.snakeyaml&type=Code

https://issues.jenkins-ci.org/browse/JENKINS-61600

**Comment by timja** [comment]
Caused https://github.com/jenkinsci/configuration-as-code-plugin/issues/1339, fixed in https://github.com/jenkinsci/configuration-as-code-plugin/pull/1340

**Comment by ipleten** [comment]
I get an error:
>org.yaml.snakeyaml.error.YAMLException: Number of aliases for non-scalar nodes exceeds the specified max=50

We have a pretty big CaSC  yaml file

**Pull request merged**